### PR TITLE
Fishermans Log UI

### PR DESCRIPTION
**Pull request by justreq**
Item: Fisherman's Log
Click while holding the item to access the UI
The item is not at all balanced or finalized, I just made it so I'm able to access the UI properly

Notes (development):
- I've created a hotkey to reset this UI for hot-reload
- Until there's a tag that stores a list of fish item types that have been caught, the UI displays all fish

### CODE REVIEW

File: Common/FishermansLogUI/FishermansLogUIState.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        searchBar = new(Language.GetText("Search"), 0.8f);
+
+        var GridScreen = MainPanel.AddElement(new FishermansLogGridScreen(grid, Fish, searchBar));
+    }
+
+    public override void Update(GameTime gameTime) {
+        Main.playerInventory = false;
+
+        if (Main.keyState.IsKeyDown(Keys.Escape) && !Main.oldKeyState.IsKeyDown(Keys.Escape)) {
+            if (searchBar.IsWritingText) searchBar.ToggleTakingText();
```
Comment (nKnaakaa) [inline Common/FishermansLogUI/FishermansLogUIState.cs:93]:
```suggestion
            if (searchBar.IsWritingText) {
                searchBar.ToggleTakingText();
            }
```

File: Common/FishermansLogUI/FishermansLogUIState.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            else {
+                SoundEngine.PlaySound(new SoundStyle("EndlessEscapade/Assets/Sounds/UI/FishermansLogClose"));
+                Visible = false;
+            }
+
+            if (Main.keyState.IsKeyDown(Keys.Enter) && !Main.oldKeyState.IsKeyDown(Keys.Enter) && searchBar.IsWritingText) searchBar.ToggleTakingText();
```
Comment (nKnaakaa) [inline Common/FishermansLogUI/FishermansLogUIState.cs:99]:
```suggestion
            if (Main.keyState.IsKeyDown(Keys.Enter) && !Main.oldKeyState.IsKeyDown(Keys.Enter) && searchBar.IsWritingText) {
                searchBar.ToggleTakingText();
            }
```

File: Common/FishermansLogUI/FishermansLogUISystem.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    public override void Unload() {
+        ResetUIKeybind = null;
+    }
+
+    private GameTime lastUpdateUiGameTime;
+
+    public override void UpdateUI(GameTime gameTime) {
+        lastUpdateUiGameTime = gameTime;
+
+        if (FishermansLogUIState.Visible) FishermansLogUserInterface.Update(gameTime);
```
Comment (nKnaakaa) [inline Common/FishermansLogUI/FishermansLogUISystem.cs:36]:
```suggestion
        if (FishermansLogUIState.Visible) {
            FishermansLogUserInterface.Update(gameTime);
        }
```

File: Common/UIElements/UIHoverItemIcon.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    Item item;
+
+    public UIHoverItemIcon(int itemType, bool blackedOut = false) : base(new Item(itemType), blackedOut) {
+        item = new Item(itemType);
+    }
+
+    public override void Draw(SpriteBatch spriteBatch) {
+        base.Draw(spriteBatch);
+
+        if (IsMouseHovering) Main.hoverItemName = item.Name;
```
Comment (nKnaakaa) [inline Common/UIElements/UIHoverItemIcon.cs:18]:
```suggestion
        if (IsMouseHovering) {
            Main.hoverItemName = item.Name;
        }
```

File: Content/Items/FishermansLog.cs [hunk 5]
```diff
@@ -1,11 +1,34 @@
...
+using Terraria.Audio;
+using Terraria.GameContent.Creative;
+using Terraria.ID;
 using Terraria.ModLoader;
 
 namespace EndlessEscapade.Content.Items;
 
 public class FishermansLog : ModItem
 {
+    public override void SetStaticDefaults() {
```
Comment (nKnaakaa) [inline Content/Items/FishermansLog.cs:12]: Remove entire method. Sacrifice counts are automatically handled from [ItemCreativeReserch.cs](https://github.com/EndlessEscapade/EndlessEscapade/blob/master/Common/Items/ItemCreativeResearch.cs)

File: Common/FishermansLogUI/FishermansLogUISystem.cs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            }, InterfaceScaleType.UI));
+        }
+    }
+}
+
+public class TemporaryPlayer : ModPlayer
+{
+    public override void ProcessTriggers(TriggersSet triggersSet) {
+        if (FishermansLogUISystem.ResetUIKeybind.JustPressed) {
+            Main.NewText("UI Reset");
```
Comment (nKnaakaa) [inline Common/FishermansLogUI/FishermansLogUISystem.cs:55]: Debugging left behind.

**Review by shnakamura — CHANGES_REQUESTED** [review]
Some issues were too recurring, so I've decided to point them separately here. They are listed down below:

* Change `public` members with `camelCase` naming to use `PascalCase` naming.
* Change redundant explicit namespaces to a `using` statement.
* Methods are placed below fields and properties.
* Custom types (structs, classes...) are placed before fields, properties and methods.

**Review by shnakamura — COMMENTED** [review]
I dont see the other mentioned changes?

**Pull request closed without merging**